### PR TITLE
Update dependencies to latest versions and resolve breaking changes

### PR DESCRIPTION
**Pull request by m-rosario**
Flutter version updated to 3.13.8 (from 3.10.1). Also updated dependencies to the latest version possible, this introduced some breaking changes (particularly in GoRouter, and lint analysis) that were also resolved.

### CODE REVIEW

**Comment by andersan** [comment]
@m-rosario everything looks good in the code changes. When I try to run the app, though, I see this error:

```
Because every version of integration_test from sdk depends on material_color_utilities 0.2.0 and mm_flutter_app depends on material_color_utilities ^0.5.0, integration_test from sdk is forbidden.
So, because mm_flutter_app depends on integration_test from sdk, version solving failed.
```

After downgrading `material_color_utilities`, there is an issue with another package:

```
flutter pub get
Resolving dependencies...
Because every version of integration_test from sdk depends on collection 1.17.1 and mm_flutter_app depends on collection 1.17.2, integration_test from sdk is forbidden.
So, because mm_flutter_app depends on integration_test from sdk, version solving failed.
```

`Integration_test`, a core flutter package, seems to cause some issues here. [These docs](https://pub.dev/packages/integration_test) explain how to include it in our app. We seem to be doing that the right way.

Are you seeing these same issues when building the app with dev dependencies?

File: pubspec.yaml [hunk 1]
```diff
@@ -55,21 +55,20 @@ dependencies:
...
   retry: ^3.1.2
   flutter_localizations:
     sdk: flutter
   firebase_messaging: ^14.6.5
   textfield_tags: ^2.0.2
   tuple: ^2.0.2
-  collection: 1.17.1
-  material_color_utilities: ^0.2.0
-  dots_indicator: ^2.1.1
+  collection: 1.17.2
```
Comment (andersan) [inline pubspec.yaml:69]:
```suggestion
  collection: ^1.17.1
```

File: pubspec.yaml [hunk 2]
```diff
@@ -55,21 +55,20 @@ dependencies:
...
+  material_color_utilities: ^0.5.0
```
Comment (andersan) [inline pubspec.yaml:70]:
```suggestion
  material_color_utilities: ^0.2.0
```

**Comment by m-rosario** [comment]
> @m-rosario everything looks good in the code changes. When I try to run the app, though, I see this error:
> 
> ```
> Because every version of integration_test from sdk depends on material_color_utilities 0.2.0 and mm_flutter_app depends on material_color_utilities ^0.5.0, integration_test from sdk is forbidden.
> So, because mm_flutter_app depends on integration_test from sdk, version solving failed.
> ```
> 
> After downgrading `material_color_utilities`, there is an issue with another package:
> 
> ```
> flutter pub get
> Resolving dependencies...
> Because every version of integration_test from sdk depends on collection 1.17.1 and mm_flutter_app depends on collection 1.17.2, integration_test from sdk is forbidden.
> So, because mm_flutter_app depends on integration_test from sdk, version solving failed.
> ```
> 
> `Integration_test`, a core flutter package, seems to cause some issues here. [These docs](https://pub.dev/packages/integration_test) explain how to include it in our app. We seem to be doing that the right way.
> 
> Are you seeing these same issues when building the app with dev dependencies?

Strange... did you update the Flutter version to 3.13.8? I believe that is something each developer has to do on their environment

**Review by andersan — APPROVED** [review]
Good call, after running `flutter upgrade` everything works fine. 

Thanks, Miguel!

**Pull request merged**